### PR TITLE
DT-2952: ⬆️  Upgrade to latest spring boot and kotlin

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -117,6 +118,6 @@ test.dependsOn testIntegration, testWithSchemaNomis
 
 tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).configureEach {
     kotlinOptions {
-        jvmTarget = "16"
+        jvmTarget = "17"
```
Comment (petergphillips) [inline build.gradle:121]: can now go to 17 for kotlin too

File: src/test/java/uk/gov/justice/hmpps/prison/service/BookingServiceTest.java [hunk 2]
```diff
@@ -92,7 +92,6 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 
-import static com.nhaarman.mockitokotlin2.MatchersKt.eq;
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/hmpps/prison/service/BookingServiceTest.java:95]: accidental import of kotlin `eq`

File: src/main/java/uk/gov/justice/hmpps/prison/service/UserService.java [hunk 3]
```diff
@@ -128,7 +128,7 @@ private void addDpsCaseloadToUser(final String username) {
         userCaseloadRepository.save(UserCaseload.builder().id(UserCaseloadId.builder().username(username).caseload(apiCaseloadId).build()).startDate(LocalDate.now()).build());
     }
 
-    @PreAuthorize("hasAnyRole('MAINTAIN_ACCESS_ROLES,MAINTAIN_ACCESS_ROLES_ADMIN')")
+    @PreAuthorize("hasAnyRole('MAINTAIN_ACCESS_ROLES','MAINTAIN_ACCESS_ROLES_ADMIN')")
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/service/UserService.java:131]: not sure how this ever worked, syntax was incorrect

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -37,6 +37,10 @@ spring:
     serialization:
       WRITE_DATES_AS_TIMESTAMPS: false
 
+  mvc:
+    pathmatch:
+      matching-strategy: ant_path_matcher
```
Comment (petergphillips) [inline src/main/resources/application.yml:42]: needed for spring fox

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderMovementResourceApiTest_moveToCell.java [hunk 5]
```diff
@@ -101,14 +101,6 @@ public void bookingId_notFound() {
         assertThat(getBodyAsJsonContent(response)).extractingJsonPathStringValue("$.userMessage").isEqualTo(bookingNotFoundError);
     }
 
-    @Test
-    public void bookingId_missing_notFound() {
-        final var response = testRestTemplate.exchange("/api/bookings//living-unit/LEI-A-1-1?reasonCode=ADM&dateTime=2020-03-24T13:24:35", PUT, anEntity(), String.class);
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderMovementResourceApiTest_moveToCell.java:106]: this was then defaulting to the global error handler so not actually calling this endpoint.  Pointless test.

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderMovementResourceApiTest_moveToCell.java [hunk 6]
```diff
@@ -135,14 +127,6 @@ public void livingUnitId_inactive_badRequest() {
         assertThat(getBodyAsJsonContent(response)).extractingJsonPathStringValue("$.userMessage").isEqualTo(livingUnitNotFoundError);
     }
 
-    @Test
-    public void livingUnitId_missing_notFound() {
-        final var response = testRestTemplate.exchange("/api/bookings/1/living-unit/?reasonCode=ADM&dateTime=2020-03-24T13:24:35", PUT, anEntity(), String.class);
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderMovementResourceApiTest_moveToCell.java:140]: this was then defaulting to the global error handler so not actually calling this endpoint.  Pointless test.

File: src/test/resources/features/locations.feature [hunk 7]
```diff
@@ -25,10 +25,6 @@ Feature: Locations
     When a request is made to retrieve location with locationId of "-9999"
     Then resource not found response is received from locations API
 
-  Scenario: Request for list/group that does not exist
```
Comment (petergphillips) [inline src/test/resources/features/locations.feature:28]: this wasn't testing the endpoint as it didn't exist, more of a test for the global not found handler which is not required.

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**